src/Packages.php

```diff
@@ -68,6 +68,16 @@ protected static function load_packages() {
 			}
 			call_user_func( array( $package_class, 'init' ) );
 		}
+
+		// Proxies "activated_plugin" hook for embedded packages listen on WC plugin activation
+		// https://github.com/woocommerce/woocommerce/issues/28697.
+		if ( is_admin() ) {
+			$is_woocommerce_just_activated = get_transient( 'woocommerce_activated_plugin' );
+			if ( $is_woocommerce_just_activated ) {
+				do_action( 'woocommerce_activated_plugin', 'woocommerce/woocommerce.php' );
+				delete_transient( 'woocommerce_activated_plugin' );
```

Comment:
perhaps we should delete transient before calling do_action to reduce chances of race conditions when actions hooked here takes lots of time. wdyt?

Reply:
That's a great suggestion, it was likely to run into a race condition. Thanks for suggesting, changes done in https://github.com/woocommerce/woocommerce/pull/28698/commits/7f76d91b07525ab564bee853ed07132f8768e8f3